### PR TITLE
Move LibraryStructureService to Jellyfin.Api

### PR DESCRIPTION
**Pull request by Ullmie02**
**Issues**
part of #2872

### CODE REVIEW

**Comment by dkanada** [comment]
Azure is working again so you might want to pull master so future pull requests will pass.

File: Jellyfin.Api/Controllers/LibraryStructureController.cs [hunk 1]
```diff
@@ -0,0 +1,347 @@
+#pragma warning disable CA1801
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/LibraryStructureController.cs:1]:
```suggestion
#nullable enable
#pragma warning disable CA1801
```

File: Jellyfin.Api/Controllers/LibraryStructureController.cs [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+            [FromQuery] LibraryOptions libraryOptions)
+        {
+            libraryOptions ??= new LibraryOptions();
+
+            if (paths != null && paths.Length > 0)
+            {
+                libraryOptions.PathInfos = paths.Select(i => new MediaPathInfo { Path = i }).ToArray();
+            }
+
+            _libraryManager.AddVirtualFolder(name, collectionType, libraryOptions, refreshLibrary);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/LibraryStructureController.cs:90]:
```suggestion
            await _libraryManager.AddVirtualFolder(name, collectionType, libraryOptions, refreshLibrary).ConfigureAwait(false);
```

Old method returns the task to be awaited elsewhere

File: Jellyfin.Api/Controllers/LibraryStructureController.cs [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+        /// <param name="refreshLibrary">Whether to refresh the library.</param>
+        /// <response code="204">Folder removed.</response>
+        /// <returns>A <see cref="NoContentResult"/>.</returns>
+        [HttpDelete]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public ActionResult RemoveVirtualFolder(
+            [FromQuery] string name,
+            [FromQuery] bool refreshLibrary)
+        {
+            _libraryManager.RemoveVirtualFolder(name, refreshLibrary);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/LibraryStructureController.cs:108]:
```suggestion
            await _libraryManager.RemoveVirtualFolder(name, refreshLibrary).ConfigureAwait(false);
```

Old method returns the task to be awaited elsewhere

**Pull request merged**